skrf/media/media.py

```diff
@@ -808,9 +810,12 @@ def line(self, d: NumberLike, unit: str = 'deg',
         unit : ['deg','rad','m','cm','um','in','mil','s','us','ns','ps']
                 the units of d.  See :func:`to_meters`, for details
         z0 : number, string, or array-like or None
-            the characteristic impedance of the line, if different
+            the port impedance of the line, if different
```

Comment:
I think this change causes too many issues with the existing code, after this change old code gives different answers. Not every user is going to follow skrf development and is not going to be aware of this change. I really want to avoid breaking backwards compatibility. `z0` should still be the characteristic impedance and port impedance should have a different name.

Reply:
Thanks for your feedback @Ttl , it is much appreciated!

This draft is intended as an exploratory work, to see how scikit-rf would be affected by the changes proposed in #651 , it will probably never be merged but enable others to chime in and provide feedback.

Here are first hands statements that I will later post on the issue:
A good example is found in test_calibration.py. A waveguide media is defined here with parameter `z0`.
https://github.com/scikit-rf/scikit-rf/blob/ac3aedfe0228615ef10da3e25212c7908059e6e0/skrf/calibration/tests/test_calibration.py#L25

Tough a dimension is specified, the physical characteristic impedance is never used (no renormalization or mismatch, the `Z0` is just like overwritten by parameter `z0`.

This media (or the losseless variant) is later used to define calibration standards like
https://github.com/scikit-rf/scikit-rf/blob/ac3aedfe0228615ef10da3e25212c7908059e6e0/skrf/calibration/tests/test_calibration.py#L389-L394

Here media `z0` is used for port and characteristic impedance. and the physical dimensions are only used to compute eventually gamma and losses. If we change the `line` behaviour, it will not change the `match`, `short`, `open`, etc. only `thru` or `attenuator` or `delay_something` would, which I think will increase confusion. `match` has even a `z_norm` booleean which is like `embed` for `line`.

Should we apply the table described in #651, we would have to specify `Z0=None` at `line`, `thru`, `attenuator` and `delay_something` which seems ugly.